components/SearchBar/index.js

```diff
@@ -0,0 +1,201 @@
+import { useState } from "react";
+import TransactionCard from "../TransactionCard/index";
+import {
+  StyledFilterButton,
+  StyledFilterContainer,
+  StyledFilterItem,
+  StyledSearchAndFilter,
+  StyledSearchContainer,
+  StyledSearchInput,
+  StyledSearchInputSelect,
+  StyledSearchUl,
+} from "./SearchBar.styled";
+
+function SearchBar({ transactions, search, onSearch, isSearchEntry }) {
+  const [searchNameValue, setSearchNameValue] = useState(search.name);
```

Comment:
I think you don't need this state and you can use instead search.name directly in your search field.

Reply:
We need the state to do a controlled check.We'll see if we can solve this differently with a database